packages/presentation/lib/ui/movie_details/bloc/movie_details_bloc.dart

```diff
@@ -1,34 +1,51 @@
+import 'package:domain/use_case/request_details_use_case.dart';
 import 'package:presentation/base/bloc.dart';
 import 'package:presentation/ui/movie_details/data/movie_details_screen_data.dart';
 import 'package:presentation/ui/movie_details/details_arguments/movie_details_arguments.dart';
 
 abstract class MovieDetailsBloc
     extends Bloc<MovieDetailsArguments, MovieDetailsScreenData> {
-  factory MovieDetailsBloc() => MovieDetailsBlocImpl();
+  factory MovieDetailsBloc(
+    RequestDetailsUseCase requestDetailsUseCase,
+  ) =>
+      _MovieDetailsBlocImpl(
+        requestDetailsUseCase,
+      );
 
   void onTapBackArrow();
 }
 
-class MovieDetailsBlocImpl
+class _MovieDetailsBlocImpl
     extends BlocImpl<MovieDetailsArguments, MovieDetailsScreenData>
     implements MovieDetailsBloc {
   MovieDetailsScreenData _screenData = MovieDetailsScreenData();
+  final RequestDetailsUseCase _detailsUseCase;
 
-  MovieDetailsBlocImpl();
+  _MovieDetailsBlocImpl(
+    this._detailsUseCase,
+  );
 
   @override
-  void initArgs(MovieDetailsArguments arguments) {
+  void initArgs(MovieDetailsArguments arguments) async {
     super.initArgs(arguments);
-    _screenData = MovieDetailsScreenData(movie: arguments.movie);
+    final id = arguments.movie.ids;
+    final response = await _detailsUseCase(id!);
+    _screenData = MovieDetailsScreenData(
+      movie: arguments.movie,
```

Comment:
Optional. U can show movie details as soon as u came to this page. What about cast? just use shimmer instead of cast list wile waiting data.